### PR TITLE
Update Dependencies

### PR DESCRIPTION
**Pull request by mmorel-35**
Supersedes #933
This pr updates gomod and github-actions dependencies
Cachés go and python dependencies. 
I have used dependabot to do that, but can cancel remove it's config if it's undesired

### CODE REVIEW

**Comment by mmorel-35** [comment]
I'm surprised that it is failing ! See https://github.com/mmorel-35/kube-bench/actions/runs/1073762386

File: .github/workflows/build.yml
Comment (yoavrotems) [inline .github/workflows/build.yml:27]: I think cache is unnecessary in this project

File: .github/workflows/build.yml
Comment (yoavrotems) [inline .github/workflows/build.yml:40]: better use major release so we won't have to update each minor release

File: .github/workflows/mkdocs-deploy.yaml
Comment (yoavrotems) [inline .github/workflows/mkdocs-deploy.yaml:23]: ditto, seems unnecessary to me in this project

File: .github/workflows/publish.yml
Comment (yoavrotems) [inline .github/workflows/publish.yml:44]: ditto about major releases

**Review by yoavrotems — CHANGES_REQUESTED** [review]

File: .github/workflows/publish.yml
Comment (mmorel-35) [reply .github/workflows/publish.yml:44]:
```suggestion
        uses: crazy-max/ghaction-docker-meta@v3
```

File: .github/workflows/build.yml
Comment (mmorel-35) [reply .github/workflows/build.yml:40]:
```suggestion
        uses: codecov/codecov-action@v2
```

File: go.mod
Comment (mmorel-35) [inline go.mod:25]: There is now a v0.21.3 for k8s.io/api , k8s.io/apimachinery and k8s.io/client-go but it requires a lot more adaptations

Comment (yoavrotems) [reply go.mod:25]: I think then that we should wait for a major release, right now its working and I don't see the value for updating it if it will requires a lot more adaptations. 
Lets cross this bridge when they will release version 1

**Comment by yoavrotems** [comment]
Hey thanks for taking care of my notes, I just updated a bit our pipeline (because kind old version was breaking stuff there haha kind of make the point for your PR) 
I would like you to update the PR and see there isn't any conflicts after this I don't see a reason not to approve it 😃

**Review by yoavrotems — APPROVED** [review]
Great! LGTM thank you for contributing

**Pull request merged**
